baseplate/context/redis.py

```diff
@@ -148,3 +151,62 @@ def __init__(self, trace_name, server_span, connection_pool,
     def execute(self, **kwargs):
         with self.server_span.make_child(self.trace_name):
             return super(MonitoredRedisPipeline, self).execute(**kwargs)
+
+
+class MessageQueue(object):
+    """A redis-backed variant of ``message_queue.MessageQueue``.
+
+    ``name`` can be any string.
+
+    ``client`` should be a ``redis.ConnectionPool`` or
+    ``redis.BlockingConnectionPool`` from which a client connection can be
+    created from (preferrably generated from the ``pool_from_config`` helper).
+
+    """
+    def __init__(self, name, client):
+        self.queue = name
+        if isinstance(client, redis.BlockingConnectionPool) \
+                or isinstance(client, redis.ConnectionPool):
+            self.client = redis.Redis(connection_pool=client)
+        else:
+            self.client = client
+
+    def get(self, timeout=None):
+        """Read a message from the queue.
+
+        :param int timeout: If the queue is empty, the call will block up to
+            ``timeout`` seconds or forever if ``None``, if a float is given,
+            it will be rounded up to be an integer
+        :raises: :py:exc:`TimedOutError` The queue was empty for the allowed
+            duration of the call.
+
+        """
+        if isinstance(timeout, float):
+            timeout = int(ceil(timeout))
+
+        if timeout == 0:
+            message = self.client.lpop(self.queue)
+        else:
+            message = self.client.blpop(self.queue, timeout=timeout)
+
+        if not message:
+            raise message_queue.TimedOutError
+
+        return message[1] if isinstance(message, tuple) else message
```

Comment:
aw, i rather liked how you'd moved this into the branches above before in the pickle version